src/frontend/src/handler/create_source.rs

```diff
@@ -1554,6 +1557,21 @@ pub async fn bind_create_source_or_table_with_connector(
     if is_create_source {
         // must behind `handle_addition_columns`
         check_and_add_timestamp_column(&with_properties, &mut columns);
+
+        // For shared sources, we will include partition and offset cols in the SourceExecutor's *output*, to be used by the SourceBackfillExecutor.
+        // For shared CDC source, the schema is different. See debezium_cdc_source_schema, CDC_BACKFILL_TABLE_ADDITIONAL_COLUMNS
+        if is_shared_non_cdc {
+            let (columns_exist, additional_columns) = source_add_partition_offset_cols(
+                &columns,
+                &with_properties.get_connector().unwrap(),
+                true, // col_id filled below at col_id_gen.generate
+            );
+            for (existed, c) in columns_exist.into_iter().zip_eq_fast(additional_columns) {
+                if !existed {
+                    columns.push(ColumnCatalog::hidden(c));
+                }
+            }
+        }
```

Comment:
I'm thinking that previously due to the logic in the planner, shared CDC source will also emit columns `_rw_mysql-cdc_offset`, `_rw_mysql-cdc_partition` (besides `_rw_table_name` and `_rw_offset`).

.. Or maybe somehow not included for CDC source. Since at the beginning of this PR, (i.e., using `is_shared`, instead of `is_shared_non_cdc` here), CDC will actually fail with index out of range at Java side..

I'm not 100% sure how it works. But it seems current stage is working.